[RemoteTestNG] detected TestNG version 7.4.0
org.testng.TestNGException: 
Cannot instantiate class APITests.TestNG.Booking.TestBookingNew
	at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:30)
	at org.testng.internal.InstanceCreator.instantiateUsingDefaultConstructor(InstanceCreator.java:193)
	at org.testng.internal.InstanceCreator.createInstanceUsingObjectFactory(InstanceCreator.java:113)
	at org.testng.internal.InstanceCreator.createInstance(InstanceCreator.java:79)
	at org.testng.internal.objects.SimpleObjectDispenser.dispense(SimpleObjectDispenser.java:25)
	at org.testng.internal.objects.GuiceBasedObjectDispenser.dispense(GuiceBasedObjectDispenser.java:30)
	at org.testng.internal.ClassImpl.getDefaultInstance(ClassImpl.java:112)
	at org.testng.internal.ClassImpl.getInstances(ClassImpl.java:165)
	at org.testng.TestClass.getInstances(TestClass.java:122)
	at org.testng.TestClass.initTestClassesAndInstances(TestClass.java:102)
	at org.testng.TestClass.init(TestClass.java:94)
	at org.testng.TestClass.<init>(TestClass.java:59)
	at org.testng.TestRunner.initMethods(TestRunner.java:463)
	at org.testng.TestRunner.init(TestRunner.java:339)
	at org.testng.TestRunner.init(TestRunner.java:292)
	at org.testng.TestRunner.<init>(TestRunner.java:223)
	at org.testng.remote.support.RemoteTestNG6_12$1.newTestRunner(RemoteTestNG6_12.java:33)
	at org.testng.remote.support.RemoteTestNG6_12$DelegatingTestRunnerFactory.newTestRunner(RemoteTestNG6_12.java:66)
	at org.testng.ITestRunnerFactory.newTestRunner(ITestRunnerFactory.java:55)
	at org.testng.SuiteRunner$ProxyTestRunnerFactory.newTestRunner(SuiteRunner.java:659)
	at org.testng.SuiteRunner.init(SuiteRunner.java:173)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:107)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1300)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1276)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1125)
	at org.testng.TestNG.runSuites(TestNG.java:1063)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:23)
	... 29 more
Caused by: java.lang.Error: Unresolved compilation problems: 
	The import APITests.TestNG.Common cannot be resolved
	APITestNGCommon cannot be resolved to a type

	at APITests.TestNG.Booking.TestBookingNew.<init>(TestBookingNew.java:10)
	... 35 more


https://cacwds--v1sit.sandbox.lightning.force.com/services/oauth2/authorize?response_type=code&client_id=3MVG9BiWFlgZEySWjz_weczaxR5Y7tquQH16FrBh.ZLCUq9CMtAax1_WmZg6X.CxCR3jTbABMVIr9ru8VuVP3&redirect_uri=https://localhost.com

APITestNGCommon
package api_tests.testng.common;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.io.File;

import org.testng.annotations.AfterSuite;
import org.testng.annotations.BeforeSuite;

import APIUtilities.APIHelpers.APIController;
import ReportUtilities.Common.ReportCommon;
import InitializeScripts.InitializeTestSettings;
import ReportUtilities.Model.TestCaseParam;
import ReportUtilities.Model.ExtentModel.PageDetails;
import TestSettings.TestRunSettings;
import APIUtilities.Models.APIReportModel;
import APIUtilities.TestSettings.APITestSettings;
import APIUtilities.Reports.ExtentReport;

public class APITestNGCommon {

	
	@BeforeSuite
    public static void testrunsetup() throws Exception
    {
	
		  Path currentRelativePath = Paths.get(""); 
		  String prjPath=currentRelativePath.toAbsolutePath().toString();

		  
			InitializeTestSettings initObj = new InitializeTestSettings();
			initObj.LoadConfigData(prjPath);
			initObj.LoadAPIConfig(prjPath);

			
			ExtentReport extentReport = new ExtentReport();
			extentReport.StartReport(TestRunSettings.ResultsPath,"API", TestRunSettings.Environment,"API");
		
    }
	
	
	public void executeAPI(TestCaseParam testCaseParam,  String apiname,String apimodulename,String apifilename) throws Exception
    {

		String filePath = APITestSettings.apiTestSettings.APIDirectory + File.separator + apifilename;

		APIController apiController = new APIController();
		ArrayList<APIReportModel> reportData=apiController.ExecuteAPI(testCaseParam.TestCaseName,apimodulename,apiname,testCaseParam.Browser,String.valueOf(testCaseParam.Iteration),filePath);


	         for(APIReportModel apiReportModel : reportData)
	         {
	        	 
	        	 PageDetails pageDetails = new PageDetails();
	        	 pageDetails.PageActionName=apiname;
	        	 pageDetails.PageActionDescription=apiname;
	        	 

					ReportCommon testStepDetails = new ReportCommon();
					testStepDetails.logAPIDetails( testCaseParam,  apiname, apiname,pageDetails, LocalDateTime.now(), apiReportModel.TestStepResult,apiReportModel.URL,apiReportModel.Request, apiReportModel.ActualResponse);

	         }

    }
	
	
	@AfterSuite
	public void endTestSuite()
	{
		ExtentReport extentReport = new ExtentReport();
		extentReport.EndReport();
	}
}

[RemoteTestNG] detected TestNG version 7.4.0
org.testng.TestNGException: 
Cannot find class in classpath: api_tests.testng.booking.TestBookingNew
	at org.testng.xml.XmlClass.loadClass(XmlClass.java:75)
	at org.testng.xml.XmlClass.init(XmlClass.java:67)
	at org.testng.xml.XmlClass.<init>(XmlClass.java:54)
	at org.testng.xml.TestNGContentHandler.startElement(TestNGContentHandler.java:576)
	at org.apache.xerces.parsers.AbstractSAXParser.startElement(Unknown Source)
	at org.apache.xerces.parsers.AbstractXMLDocumentParser.emptyElement(Unknown Source)
	at org.apache.xerces.impl.dtd.XMLDTDValidator.emptyElement(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanStartElement(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl.parse(Unknown Source)
	at java.xml/javax.xml.parsers.SAXParser.parse(SAXParser.java:197)
	at org.testng.xml.XMLParser.parse(XMLParser.java:39)
	at org.testng.xml.SuiteXmlParser.parse(SuiteXmlParser.java:16)
	at org.testng.xml.SuiteXmlParser.parse(SuiteXmlParser.java:9)
	at org.testng.xml.Parser.parse(Parser.java:164)
	at org.testng.xml.Parser.parse(Parser.java:250)
	at org.testng.TestNG.parseSuite(TestNG.java:324)
	at org.testng.TestNG.initializeSuitesAndJarFile(TestNG.java:376)
	at org.testng.TestNG.initializeEverything(TestNG.java:1011)
	at org.testng.remote.support.RemoteTestNG6_12.initialize(RemoteTestNG6_12.java:22)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:98)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)


public void processLogin(TestCaseParam testCaseParam, String iterationNum) throws IOException, InterruptedException  {
 
		PageDetails action = new PageDetails();
		LocalDateTime startTime= LocalDateTime.now();
		action.setPageActionName("Process Login");
		action.setPageActionDescription("Process Login");
 
		try {			
			ExcelUtils.loadDataForTestScript("Login", TestRunSettings.getTestNGTestMethodName() ,iterationNum);
			String userName = TestRunSettings.getExcelValue("LOGINUSER");
			
			String[] credentials = SalesforceConstants.getUserCredentials(userName);
 
			Webkeywords.instance().navigate(driver,  TestRunSettings.getUrl(), testCaseParam,action);
			Webkeywords.instance().setText(driver, txtUserName, credentials[0], testCaseParam,action);
			Webkeywords.instance().setText(driver, txtPassword, credentials[1], testCaseParam,action);
			
			Webkeywords.instance().click(driver, btnLoginButton,"", testCaseParam,action);
		
			Webkeywords.instance().waitElementToBeVisibleNew(driver, txtcares, 2);
			Webkeywords.instance().click(driver, txtcares, "", testCaseParam,action);
 
			objcommonOperations.switchToLastTab(driver);
			Webkeywords.instance().waitTitleContains(driver, "Home | Salesforce");
			driver.switchTo().defaultContent();
		}
		catch (IOException e)
		{
			logger.error("Failed == {} ", action.getPageActionDescription());
			exceptionDetails.logExceptionDetails(driver, testCaseParam, action.getPageActionName(), action.getPageActionDescription(), startTime,e);
		} catch (FilloException e) {
 
			logger.error(e.getMessage());
		}
has context menu

----------------------------------------------------
public void processLogin(TestCaseParam testCaseParam, String iteration) throws IOException, InterruptedException  {
 
		PageDetails action = new PageDetails();
		LocalDateTime startTime= LocalDateTime.now();
		action.setPageActionName("Process Login");
		action.setPageActionDescription("Process Login");
		try {
			Map<String, ArrayList<String>> testcaseDataSD;
			testcaseDataSD = util.getScreenTCData(screenName, testCaseParam.getTestCaseName(),TestRunSettings.getTestDataPath(), TestRunSettings.getTestDataMappingFileName() ,TestRunSettings.getTestDataMappingSheetNameSd(),iteration);
 
			String url = testcaseDataSD.get("URL").get(0);
			String userName = testcaseDataSD.get("UserName").get(0);
 
			String password = testcaseDataSD.get("Password").get(0);
			String submitButton = testcaseDataSD.get("SubmitButton").get(0);
 
			Webkeywords.instance().navigate(driver,  url, testCaseParam,action);
			Webkeywords.instance().setText(driver, txtUserName, userName, testCaseParam,action);
			Webkeywords.instance().setText(driver, txtPassword, password, testCaseParam,action);
			Webkeywords.instance().click(driver, btnLoginButton,submitButton, testCaseParam,action);
		
			Webkeywords.instance().waitElementToBeVisibleNew(driver, selectApp, 2);
			objcommonOperations.pageRefresh(driver);
			Webkeywords.instance().click(driver, selectApp,"", testCaseParam,action);
			
			objcommonOperations.switchToLastTab(driver);
			driver.switchTo().defaultContent();
		}
		catch (IOException e)
		{
			logger.error("Failed == {} ", action.getPageActionDescription());
			exceptionDetails.logExceptionDetails(driver, testCaseParam, action.getPageActionName(), action.getPageActionDescription(), startTime,e);
		}
	}

username: input[@id='input27']  password: input[@id='input35']
